### DISCUSSION TITLE
get help

### DISCUSSION THREAD

**Discussion by nowrozidavoud** [ConsenSys-archive/ganache#4363] [Q&A] [answered]
Hello dears;
I am a widows 10 user.
Nowadays start my journey in web3 (ethereum) and need to work by ganache-cli.
First install WSL in my system and by powershell install ubuntu 16.04 LTS.
Run this command :
**david@DESKTOP-Q17APUQ:~$ npm install -g ganache-cli**
and get the result :
npm WARN deprecated ganache-cli@6.12.2: ganache-cli is now ganache; visit https://trfl.io/g7 for details
npm WARN checkPermissions Missing write access to /usr/local/lib/node_modules
/usr/local/lib
└─┬ ganache-cli@6.12.2
  ├─┬ ethereumjs-util@6.2.1
  │ ├─┬ @types/bn.js@4.11.6
  │ │ └── @types/node@14.11.2
  │ ├── bn.js@4.11.9
  │ ├─┬ create-hash@1.2.0
  │ │ ├── cipher-base@1.0.4
  │ │ ├── inherits@2.0.4
  │ │ ├─┬ md5.js@1.3.5
  │ │ │ └─┬ hash-base@3.1.0
  │ │ │   └─┬ readable-stream@3.6.0
  │ │ │     ├── string_decoder@1.3.0
  │ │ │     └── util-deprecate@1.0.2
  │ │ ├── ripemd160@2.0.2
  │ │ └── sha.js@2.4.11
  │ ├─┬ elliptic@6.5.3
  │ │ ├── brorand@1.1.0
  │ │ ├── hash.js@1.1.7
  │ │ ├── hmac-drbg@1.0.1
  │ │ ├── minimalistic-assert@1.0.1
  │ │ └── minimalistic-crypto-utils@1.0.1
  │ ├─┬ ethereum-cryptography@0.1.3
  │ │ ├── @types/pbkdf2@3.1.0
  │ │ ├── @types/secp256k1@4.0.1
  │ │ ├── blakejs@1.1.0
  │ │ ├─┬ browserify-aes@1.2.0
  │ │ │ ├── buffer-xor@1.0.3
  │ │ │ └── evp_bytestokey@1.0.3
  │ │ ├─┬ bs58check@2.1.2
  │ │ │ └─┬ bs58@4.0.1
  │ │ │   └── base-x@3.0.8
  │ │ ├── create-hmac@1.1.7
  │ │ ├─┬ keccak@3.0.1
  │ │ │ ├── node-addon-api@2.0.2
  │ │ │ └── node-gyp-build@4.2.3
  │ │ ├── pbkdf2@3.1.1
  │ │ ├── randombytes@2.1.0
  │ │ ├── safe-buffer@5.2.1
  │ │ ├── scrypt-js@3.0.1
  │ │ ├── secp256k1@4.0.2
  │ │ └── setimmediate@1.0.5
  │ ├─┬ ethjs-util@0.1.6
  │ │ ├── is-hex-prefixed@1.0.0
  │ │ └── strip-hex-prefix@1.0.0
  │ └── rlp@2.2.6
  ├─┬ source-map-support@0.5.12
  │ ├── buffer-from@1.1.1
  │ └── source-map@0.6.1
  └─┬ yargs@13.2.4
    ├─┬ cliui@5.0.0
    │ ├─┬ strip-ansi@5.2.0
    │ │ └── ansi-regex@4.1.0
    │ └─┬ wrap-ansi@5.1.0
    │   └─┬ ansi-styles@3.2.1
    │     └─┬ color-convert@1.9.3
    │       └── color-name@1.1.3
    ├─┬ find-up@3.0.0
    │ └─┬ locate-path@3.0.0
    │   ├─┬ p-locate@3.0.0
    │   │ └─┬ p-limit@2.3.0
    │   │   └── p-try@2.2.0
    │   └── path-exists@3.0.0
    ├── get-caller-file@2.0.5
    ├─┬ os-locale@3.1.0
    │ ├─┬ execa@1.0.0
    │ │ ├─┬ cross-spawn@6.0.5
    │ │ │ ├── nice-try@1.0.5
    │ │ │ ├── path-key@2.0.1
    │ │ │ ├── semver@5.7.1
    │ │ │ ├─┬ shebang-command@1.2.0
    │ │ │ │ └── shebang-regex@1.0.0
    │ │ │ └─┬ which@1.3.1
    │ │ │   └── isexe@2.0.0
    │ │ ├─┬ get-stream@4.1.0
    │ │ │ └─┬ pump@3.0.0
    │ │ │   ├── end-of-stream@1.4.4
    │ │ │   └─┬ once@1.4.0
    │ │ │     └── wrappy@1.0.2
    │ │ ├── is-stream@1.1.0
    │ │ ├── npm-run-path@2.0.2
    │ │ ├── p-finally@1.0.0
    │ │ ├── signal-exit@3.0.3
    │ │ └── strip-eof@1.0.0
    │ ├─┬ lcid@2.0.0
    │ │ └── invert-kv@2.0.0
    │ └─┬ mem@4.3.0
    │   ├─┬ map-age-cleaner@0.1.3
    │   │ └── p-defer@1.0.0
    │   ├── mimic-fn@2.1.0
    │   └── p-is-promise@2.1.0
    ├── require-directory@2.1.1
    ├── require-main-filename@2.0.0
    ├── set-blocking@2.0.0
    ├─┬ string-width@3.1.0
    │ ├── emoji-regex@7.0.3
    │ └── is-fullwidth-code-point@2.0.0
    ├── which-module@2.0.0
    ├── y18n@4.0.0
    └─┬ yargs-parser@13.1.2
      ├── camelcase@5.3.1
      └── decamelize@1.2.0

npm ERR! Linux 4.4.0-19041-Microsoft
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "install" "-g" "ganache-cli"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.2
npm ERR! path /usr/local/lib/node_modules
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall access

npm ERR! Error: EACCES: permission denied, access '/usr/local/lib/node_modules'
npm ERR!     at Error (native)
npm ERR!  { [Error: EACCES: permission denied, access '/usr/local/lib/node_modules']
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/usr/local/lib/node_modules' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/david/npm-debug.log

These is 2 days I stoped here and search for it and do many command but my problem is not solve yet.
Any one can help me please?

**Comment by jeffsmale90** [comment, marked as answer]
The `ganache-cli` package is deprecated. Please try `npm install -g ganache`.

This probably won't resolve your problem, which appears to be a permissions issue with the location where npm installs global packages. If the above doesn't help, try following the instructions from npm here https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally.
